### PR TITLE
feat: introducing tags for conditional forwarding

### PR DESCRIPTION
**Pull request by yhl25**

### CODE REVIEW

**Review by vigith — APPROVED** [review]

File: pynumaflow/function/_dtypes.py [hunk 1]
```diff
@@ -103,19 +106,33 @@ class MessageT(metaclass=NoPublicConstructor):
...
     _value: bytes = b""
     # There is no year 0, so setting following as default event time.
     _event_time: datetime = datetime(1, 1, 1, 0, 0)
 
-    @classmethod
-    def to_vtx(cls: Type[MT], keys: List[str], value: bytes, event_time: datetime) -> MT:
+    def __init__(self, value: bytes):
         """
-        Returns a MessageT object to send value to a vertex.
+        Returns a Message object to send value to a vertex.
```
Comment (ab93) [inline pynumaflow/function/_dtypes.py:116]: Shouldn't this be `MessageT`?

File: pynumaflow/function/_dtypes.py [hunk 2]
```diff
@@ -28,17 +26,27 @@ class Message(metaclass=NoPublicConstructor):
     """
 
     _keys: List[str]
+    _tags: List[str]
     _value: bytes = b""
 
-    @classmethod
-    def to_vtx(cls: Type[M], keys: List[str], value: bytes) -> M:
+    def __init__(self, value: bytes):
```
Comment (ab93) [inline pynumaflow/function/_dtypes.py:32]: @vigith any reason why we are not allowing users to initialize the object from the constructor itself?

**Review by ab93 — COMMENTED** [review]
Looks good, just a few clarifying comments

File: pynumaflow/function/_dtypes.py [hunk 3]
```diff
@@ -74,24 +86,15 @@ def append(self, message: Message) -> None:
...
     def dumps(self) -> str:
         return str(self)
 
     def loads(self) -> Ms:
         pass
 
 
-@dataclass(frozen=True)
-class MessageT(metaclass=NoPublicConstructor):
+@dataclass()
```
Comment (ab93) [inline pynumaflow/function/_dtypes.py:96]:
```suggestion
@dataclass
```

File: pynumaflow/function/_dtypes.py (hunk 2 above)
Comment (yhl25) [reply pynumaflow/function/_dtypes.py]: discussed with Vigith offline, and decided to go with constructor for object creation.

**Review by vigith — APPROVED** [review]

**Review by ashwinidulams — APPROVED** [review]

**Pull request merged**